### PR TITLE
Add new middleware system and auth middleware

### PR DESCRIPTION
**Pull request by Fenntasy**
## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

File: src/middlewares/withLogger.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { ServerResponse } from "http";
+
+import { Handler } from "../@types/ApiPageHandler";
+import { ExtendedRequest } from "../@types/ExtendedRequest";
+
+const logger = createLogger({
+  service: "connect-account",
+  encoder: EncoderTypeEnum.JSON,
+}).withMeta({
+  process: "apiPage",
```
Comment (Fuzznimp) [inline src/middlewares/withLogger.ts:11]: This could also be SSR, right?

Comment (Fenntasy) [reply src/middlewares/withLogger.ts:11]: this is

Comment (Fuzznimp) [reply src/middlewares/withLogger.ts:11]: So we don't differentiate API routes from getServerSidesProps functions?

File: src/pages/account/logins/[type]/[id].tsx [hunk 2]
```diff
@@ -40,34 +42,23 @@ const IdentityOverviewPage: React.FC<{ identity: Identity }> = ({
...
-      const userDocumentId = context.req.session.get("user-session-id");
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  return wrapMiddlewaresForSSR<{ type: string }>(
+    context,
+    [withLogger, withSentry, withMongoDB, withSession, withAuth],
+    async (request: ExtendedRequest, response: ServerResponse) => {
+      if (!context?.params?.id) {
+        response.statusCode = 400;
+        response.end();
+        return;
```
Comment (Fuzznimp) [inline src/pages/account/logins/[type]/[id].tsx:53]: Any reason why you did not went with `return response.end()`?

Comment (Fenntasy) [reply src/pages/account/logins/[type]/[id].tsx:53]: psychological safety, I wanted to be sure to have a function that return void at the moment, and maybe let the possibility of doing something else (but not really no, to answer honestly)

Comment (Fuzznimp) [reply src/pages/account/logins/[type]/[id].tsx:53]: > psychological safety
I had the feeling you did it for this, and I'm more than fine with it.

**Review by Oluane — DISMISSED** [review]

**Review by Oluane — APPROVED** [review]

**Pull request merged**